src/Controller/CRUDController.php

```diff
@@ -286,20 +286,27 @@ public function deleteAction($id)
     /**
      * Edit action.
      *
-     * @param int|string|null $id
+     * @param int|string|null $deprecatedId
      *
      * @throws NotFoundHttpException If the object does not exist
      * @throws \RuntimeException     If no editable field is defined
      * @throws AccessDeniedException If access is not granted
      *
      * @return Response|RedirectResponse
      */
-    public function editAction($id = null)
+    public function editAction($deprecatedId = null) // NEXT_MAJOR: Remove the unused $id parameter
     {
-        $request = $this->getRequest();
+        if (isset(\func_get_args()[0])) {
+            @trigger_error(
+                'The $id param will be remove in 4.0, use $this->admin->getIdParameter() instead',
```

Comment:
```suggestion
                sprintf('Support for the "id" route param as argument 1 at `%s()` is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0, use `AdminInterface::getIdParameter()` instead.', __METHOD__),
```